File: Backend/BaseMgtService/Senior.AgileAI.BaseMgt.Application/DTOs/updateProfileDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Senior.AgileAI.BaseMgt.Application.DTOs
{
    public class updateProfileDTO
    {
        public string? FullName { get; set; } //the frontend can only send the updated fields.
        public DateOnly? BirthDate { get; set; }
        public Guid? CountryId { get; set; }
        // TODO: add the profile picture
    }
}

File: agilemeets_mobile/lib/data/models/profile/update_profile_dto.dart
class UpdateProfileDTO {
  final String? fullName;
  final DateTime? birthDate;
  final String? countryId;

  UpdateProfileDTO({
    this.fullName,
    this.birthDate,
    this.countryId,
  });

  Map<String, dynamic> toJson() {
    return {
      'fullName': fullName,
      'birthDate': birthDate?.toIso8601String().split('T')[0],
      'countryId': countryId,
    };
  }
} 